src/models/NodeModel.ts

```diff
@@ -71,6 +70,16 @@ export class NodeModel extends BaseModel<BaseModelListener> {
 		});
 	}
 
+	clone(lookupTable) {
+		if(((lookupTable||{})[this.class]||{}).hasOwnProperty(this.id)) return lookupTable[this.class][this.id];
```

Comment:
should this line not exist on the BaseModel since its used in all the inherited models?